Carbon/Functions/Write-File.ps1

```diff
@@ -161,10 +161,7 @@ function Write-File
         Set-StrictMode -Version 'Latest'
         Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
 
-        $msg = "The Carbon module's ""Write-File"" function was renamed to ""Write-CFile"". Please update " +
-                "your code to use the new ""Write-CFile"" name. The old ""Write-File"" function will be " +
-                'removed in the next major version of Carbon.'
-        Write-CWarningOnce -Message $msg
+        Write-CRenamedCommandWarning -CommandName $MyInvocation.MyCommand.Name -NewCommandName 'Write-CFile'

```

Comment:
The existing command name and the new command name are the same?

Reply:
No The existing command is `Write-File`.

Reply:
Does it get renamed somewhere? The function is already defined with the `C` prefix: [Write-File.ps1:2](https://github.com/webmd-health-services/Carbon/blob/develop/Carbon/Functions/Write-File.ps1#L2)

Also how do we know that the caller didn't call it with the correct name? Looks like we're writing the warning regardless of how it was called.

Reply:
None of the Carbon functions used to have the `C` prefix. That caused some problems with new commands published by Microsoft and others. So, I renamed them all to have a `C` prefix and created aliases from the old name to the new name. Unfortunately, aliases have the highest precedence, so that caused problems with people using Carbon with those other modules. Now, I dynamically define shim functions with the old name (i.e. without the `C`) that call the function with the new name (i.e. with the `C`). This shim doesn't work with `Compress-CItem` and `Write-CFile`, so I had to create those two shims by hand.

Reply:
Got it, thanks for explaining.